app/controllers/devise/two_step_verification_controller.rb

```diff
@@ -1,10 +1,18 @@
 class Devise::TwoStepVerificationController < DeviseController
-  before_filter :prepare_and_validate
+  before_filter :prepare_and_validate, except: [:prompt, :defer]
   skip_before_filter :handle_two_step_verification
 
   attr_reader :otp_secret_key
   private :otp_secret_key
 
+  def prompt
+  end
+
+  def defer
+    current_user.defer_two_step_verification
+    redirect_to root_path, notice: 'You will not be required to setup 2-step verification.'
```

Comment:
@jamiecobbett OK, I'll address these issues. Is it OK to leave deferral as a permanent thing for now? So if they defer, it will require a super-admin to flag them again. There is an audit trail via the event log recording, so perhaps it won't go unnoticed?